analytics/aggregate.go

```diff
@@ -576,272 +679,279 @@ func AggregateData(data []interface{}, trackAllPaths bool, ignoreTagPrefixList [
+			TotalRequestTime: float64(record.RequestTime),
+			LastTime:         record.TimeStamp,
+
+			MaxUpstreamLatency:   record.Latency.Upstream,
+			MinUpstreamLatency:   record.Latency.Upstream,
+			TotalUpstreamLatency: record.Latency.Upstream,
+			MaxLatency:           record.Latency.Total,
+			MinLatency:           record.Latency.Total,
+			TotalLatency:         record.Latency.Total,
+			ErrorMap:             make(map[string]int),
+		}
+		aggregate.Total.Hits++
+		aggregate.Total.TotalRequestTime += float64(record.RequestTime)
+
+		// We need an initial value
+		aggregate.Total.RequestTime = aggregate.Total.TotalRequestTime / float64(aggregate.Total.Hits)
+		if record.ResponseCode >= 400 {
+			thisCounter.ErrorTotal = 1
+			thisCounter.ErrorMap[strconv.Itoa(record.ResponseCode)]++
+			aggregate.Total.ErrorTotal++
+			aggregate.Total.ErrorMap[strconv.Itoa(record.ResponseCode)]++
+		}
 
-			if (thisV.ResponseCode < 300) && (thisV.ResponseCode >= 200) {
-				thisCounter.Success = 1
-				thisAggregate.Total.Success++
-			}
+		if (record.ResponseCode < 300) && (record.ResponseCode >= 200) {
+			thisCounter.Success = 1
+			aggregate.Total.Success++
+		}
 
-			thisAggregate.Total.TotalLatency += thisV.Latency.Total
-			thisAggregate.Total.TotalUpstreamLatency += thisV.Latency.Upstream
+		aggregate.Total.TotalLatency += record.Latency.Total
+		aggregate.Total.TotalUpstreamLatency += record.Latency.Upstream
 
-			if thisAggregate.Total.MaxLatency < thisV.Latency.Total {
-				thisAggregate.Total.MaxLatency = thisV.Latency.Total
-			}
+		if aggregate.Total.MaxLatency < record.Latency.Total {
+			aggregate.Total.MaxLatency = record.Latency.Total
+		}
 
-			if thisAggregate.Total.MaxUpstreamLatency < thisV.Latency.Upstream {
-				thisAggregate.Total.MaxUpstreamLatency = thisV.Latency.Upstream
-			}
+		if aggregate.Total.MaxUpstreamLatency < record.Latency.Upstream {
+			aggregate.Total.MaxUpstreamLatency = record.Latency.Upstream
+		}
 
-			// by default, min_total_latency will have 0 value
-			// it should not be set to 0 always
-			if thisAggregate.Total.Hits == 1 {
-				thisAggregate.Total.MinLatency = thisV.Latency.Total
-				thisAggregate.Total.MinUpstreamLatency = thisV.Latency.Upstream
-			} else {
-				// Don't update min latency in case of error
-				if thisAggregate.Total.MinLatency > thisV.Latency.Total && (thisV.ResponseCode < 300) && (thisV.ResponseCode >= 200) {
-					thisAggregate.Total.MinLatency = thisV.Latency.Total
-				}
-				// Don't update min latency in case of error
-				if thisAggregate.Total.MinUpstreamLatency > thisV.Latency.Upstream && (thisV.ResponseCode < 300) && (thisV.ResponseCode >= 200) {
-					thisAggregate.Total.MinUpstreamLatency = thisV.Latency.Upstream
-				}
+		// by default, min_total_latency will have 0 value
+		// it should not be set to 0 always
+		if aggregate.Total.Hits == 1 {
+			aggregate.Total.MinLatency = record.Latency.Total
+			aggregate.Total.MinUpstreamLatency = record.Latency.Upstream
+		} else {
```

Comment:
@kofoworola we consider the same as `graph error`? If not, we are going to be updating the data for graph requests independent of the error

Reply:
graph errors can include the latency data if that's what you mean, we could have situations where there is a graph error while the request was still "processed" with a response.

Reply:
@kofoworola it would be good to add some comments and documentation for it since it's going to behave differently from the "normal" aggregations in that aspect.

Reply:
oh so this function is the same as it was before, I just extracted it from AggregateData, so it can be used by aggregate graph data.

Reply:
@tbuchaillot